src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHelper.Digest.cs

```diff
@@ -359,18 +372,30 @@ private string GetNextValue(string data, int currentIndex, out int parsedIndex)
                     }
                 }
 
+                // Skip the quote.
+                if (quotedValue)
+                    currentIndex++;
+
+                // Skip any whitespace.
+                while (currentIndex < data.Length && CharIsSpaceOrTab(data[currentIndex]))
+                    currentIndex++;
+
                 // Return if this is last value.
                 if (currentIndex == data.Length)
                 {
                     parsedIndex = currentIndex;
                     return StringBuilderCache.GetStringAndRelease(sb);
                 }
 
-                // Skip the end quote or ',' or space or tab.
-                currentIndex++;
+                // A key-value pair should end with ','
+                if (data[currentIndex++] != ',')
+                {
+                    parsedIndex = currentIndex;
+                    return null;
```

Comment:
Is there any benefit to making our parsing more strict here? Looking at the spec I agree that it is probably correct not to generate key value pairs that end with spaces or tabs. But accepting them when we do see them might be good.